app/controllers/admin_controller.rb

```diff
@@ -181,23 +180,24 @@ def delete_results_data
   end
 
   def create_results
-    @competition = competition_from_params
+    competition = competition_from_params
 
     # Do json analysis + insert record in db, then redirect to check inbox
     # (and delete existing record if any)
-    upload_json_params = params.require(:upload_json).permit(:results_file)
-    upload_json_params[:competition_id] = @competition.id
-    @upload_json = UploadJson.new(upload_json_params)
+    upload_json = UploadJson.new({
+                                   results_file: params.require(:results_file),
+                                   competition_id: competition.id,
+                                 })
+    mark_result_submitted = ActiveRecord::Type::Boolean.new.cast(params.require(:mark_result_submitted))
 
     # This makes sure the json structure is valid!
-    if @upload_json.import_to_inbox
-      @competition.update!(results_submitted_at: Time.now) if @competition.results_submitted_at.nil?
-      flash[:success] = "JSON file has been imported."
-      redirect_to competition_admin_upload_results_edit_path
+    if upload_json.import_to_inbox
+      competition.touch(:results_submitted_at) if competition.results_submitted_at.nil? && mark_result_submitted
```

Comment:
Just curious, in what circumstance do you need to run `import_to_inbox` but then also ignore the `results_submitted_at` after submission has been successful?
In other words, when will `mark_result_submitted` be `false`?

Reply:
In normal cases (ie, when Delegates upload the JSON), `results_submitted_at` won't get changed. Even the form won't have any way to set this to true. It's because the results will be "marked as submitted", only when they fill the form.

But when WRT uploads a result through WRT's page, then the `results_submitted_at` needs to be updated. This is because WRT doesn't have to fill the form.

I understand it's not so good to have two different UIs to do the same thing, also the setting of `results_submitted_at` looks bit hacky, and I definitely have the plans to improve this. Once the migration to React is completed, there are plenty of cleanups that will help improve this.